src/subscription.rs

```diff
@@ -9,7 +9,7 @@ use std::sync::{Arc, Mutex};
 
 pub struct SubscriptionRegistry<TReq: Debug + Eq + Hash, TRes: Debug> {
     subscriptions: HashMap<TReq, Arc<Mutex<Subscription<TRes>>>>,
-    cancelled: bool,
+    shutting_down: bool,
 }
 
 impl<TReq: Debug + Eq + Hash, TRes: Debug> fmt::Debug for SubscriptionRegistry<TReq, TRes> {
```

Comment:
unable to comment on the actual lines (21, 22) but the changes to this file reminded me: the Debug formatter is broken (`std::any::type_name::<Self>()` already includes the TReq and TRes so the output has the two TReq and TRes twice.

Reply:
I seem to recall that bit; I'll include that with my next changes :+1: